### ISSUE TITLE
RSS missing feed title

### ISSUE THREAD

**Issue by AlexHedley** [statiqdev/Statiq#1012]
I was looking at using [IFTTT](https://ifttt.com/create) to post items based on an RSS feed.

I got the following error using the feed url:

http://alexhedley.com/blog/feed.rss

> Not a valid feed url, missing feed title

![image](https://github.com/statiqdev/Statiq.Web/assets/1573469/92d09fde-53b4-4ace-b5cb-2cad10510b48)

Other services like [Zapier](https://zapier.com/) have no issue.

---

Code:

https://github.com/statiqdev/Statiq.Web/blob/main/src/Statiq.Web/Pipelines/Feeds.cs

**Comment by daveaglick** [comment]
I can see the same on the W3C validation service (https://validator.w3.org/feed/check.cgi?url=http%3A%2F%2Falexhedley.com%2Fblog%2Ffeed.rss):

![image](https://github.com/statiqdev/Statiq.Web/assets/1020407/898e2496-d353-4acc-858e-20ff26a806ad)

At one point these feeds we validating, so I'm not sure if there's a regression, certain settings aren't carrying through like they're supposed to, or the validator has gotten better. In any case, let me see what I can do about cleaning it up.

**Comment by daveaglick** [comment]
The issue here is due to a mismatch between some of the variables. The `Feeds` pipeline in Statiq Web attempts to set `FeedTitle` as the title of the feed. That allows you to override the title for any given feed from the [feed YAML file](https://www.statiq.dev/guide/web/feeds). It's fine if one isn't specified though (and one is not [in the Clean Blog theme's `feed.yaml` file](https://github.com/statiqdev/CleanBlog/blob/main/input/feed.yml) for example) because the underlying `GenerateFeed` module will fallback to `Title` for the feed title, which should represent the title of the site. The problem is that Clean Blog (and probably most other themes or custom sites due to that as an example) are using `SiteTitle` instead of `Title` to set the title due to trying to establish a convention of `Title` as a local page title and `SiteTitle` as the site title, so no `Title` metadata gets set at all.

A quick fix for you would be to set `FeedTitle` globally. A longer-term fix that I'll have coming out in the next release (later this week) is to make sure the feed used `SiteTitle` if _both_ `Title` and `FeedTitle` are null or empty.

**Comment by daveaglick** [comment]
This is now validating with the unreleased changes:

![image](https://github.com/statiqdev/Statiq.Web/assets/1020407/50030296-39de-4d02-b936-93dea9627983)

I did notice one more thing that's less a bug and more a documentation issue. Feeds require a valid absolute URL to validate, so specifying `Host` setting (I.e. in the `settings.yml` file) is important for valid feeds. I've added a little warning if one isn't specified and will add some documentation about this.

Otherwise, this bug will be resolved on the next release so I'll close this issue. Thanks for reporting it!
